base/some.jl

```diff
@@ -138,8 +138,8 @@ true
 macro something(args...)
     expr = :(nothing)
     for arg in reverse(args)
-        expr = :((val = $arg) !== nothing ? val : $expr)
+        expr = :(val = $(esc(arg)); val === nothing ? $expr : val)
```

Comment:
If it's all the same can we do this ordering to improve the readability of the macro-expanded code? The parenthesis aren't necessary but I think add clarity of precedence with the multiple statements
```suggestion
        expr = :(val = $(esc(arg)); val !== nothing ? val : ($expr))
```